### PR TITLE
fix: Support for TS 4.5

### PR DESCRIPTION
**Pull request by crpc**

### CODE REVIEW

**Comment by pawfa** [comment]
Have you checked that after this merge pact-gen-ts can be used  in apps with older versions of typescript?

**Comment by crpc** [comment]
> Have you checked that after this merge pact-gen-ts can be used in apps with older versions of typescript?

No, it won't work with TS 4.2. Ts-morph v13 supports TS 4.5, and as this project is built on ts-morph, we have to follow their support of TypeScript. If someone uses older version of TS, he should stay on older version of pact-gen-ts.

**Comment by Fiszcz** [comment]
For me, looks nice.

But what do you think about this crazy idea:
Install different versions of `ts-morph` using npm/yarn aliases:
https://stackoverflow.com/a/56495651/8850253
and then use a specific version of tsMorph depending on the installed version of TypeScript (in the user project).
It would look something like that:
```ts
import * as tsMorph12 from 'tsMorph12';
import * as tsMorph13 from 'tsMorph13';

export const tsMorph = typescriptVersion >= 4.5 ? 
        {...tsMorph13, getComment: (jsDocTag) => jsDocTag.getCommentText()} 
        : {...tsMorph12, getComment: (jsDocTag) => jsDocTag.getComment()};
```
Probably, it is the most stupid idea which I have made up, but maybe ? :P

**Comment by Fiszcz** [comment]
Should we add annotation about TypeScript support in the README.md file ?

**Review by Fiszcz — APPROVED** [review]

**Comment by crpc** [comment]
> For me, looks nice.
> 
> But what do you think about this crazy idea: Install different versions of `ts-morph` using npm/yarn aliases: https://stackoverflow.com/a/56495651/8850253 and then use a specific version of tsMorph depending on the installed version of TypeScript (in the user project). It would look something like that:
> 
> ```ts
> import * as tsMorph12 from 'tsMorph12';
> import * as tsMorph13 from 'tsMorph13';
> 
> export const tsMorph = typescriptVersion >= 4.5 ? 
>         {...tsMorph13, getComment: (jsDocTag) => jsDocTag.getCommentText()} 
>         : {...tsMorph12, getComment: (jsDocTag) => jsDocTag.getComment()};
> ```
> 
> Probably, it is the most stupid idea which I have made up, but maybe ? :P

This only works on the project they are used on, if we want to use them to provide support for projects using pact-gen-ts, we can't really rely on npm aliases

**Comment by pawfa** [comment]
@crpc Regarding the "crazy idea" :) - ts-morph maintainers had the same concept implemented for typescript. Looking at the comments here: https://github.com/dsherret/ts-morph/issues/144 and here https://github.com/dsherret/ts-morph/issues/1231 - it was a maintenance nightmare. So for now I guess we have to align with typescript releases and breaking changes.

**Review by pawfa — APPROVED** [review]

File: package.json
Comment (lukaszfiszer) [inline package.json:46]: We need some section in the readme which says that version 0.x are compatible with TS 4.2 (and 4.3/4.4 probably), and that 1.x is compatible with 4.5

**Review by lukaszfiszer — CHANGES_REQUESTED** [review]

File: package.json
Comment (crpc) [reply package.json:3]: Are we ready to release v1.0.0 though? I could bump version to 0.9, patch release may in fact not reflect the compatibility breaking change

File: package.json
Comment (crpc) [reply package.json:46]: sure thing, will add some section with compatibility info to readme

File: package.json
Comment (lukaszfiszer) [reply package.json:3]: You're right, let's bump to 0.9

**Review by lukaszfiszer — APPROVED** [review]

**Pull request merged**
